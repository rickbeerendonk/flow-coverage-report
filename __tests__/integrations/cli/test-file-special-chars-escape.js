'use babel';

import path from 'path';

import {
  FIXTURE_PATH,
  runFlowCoverageReport
} from '../../common';

const testProjectDir = path.join(FIXTURE_PATH, 'file-special-chars-escape');

jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000; // 10 second timeout

test('Fixed #92 - Escape special chars in filenames', async () => {
  const {stdout, stderr} = await runFlowCoverageReport([
    '-i', `"src/*.js"`
  ], {cwd: testProjectDir});

  const filteredStdout = stdout.split('\n').filter(
    line => line.indexOf('src/file-with-a') >= 0
  );

  expect({filteredStdout, stderr}).toMatchSnapshot();
});
