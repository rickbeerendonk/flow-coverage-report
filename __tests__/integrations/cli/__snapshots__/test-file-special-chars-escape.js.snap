// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fixed #92 - Escape special chars in filenames 1`] = `
Object {
  "filteredStdout": Array [
    "│ src/file-with-a space.js  │       flow │   100 % │     3 │       3 │ 0         │",
    "│ src/file-with-a-\\"-char.js │       flow │   100 % │     3 │       3 │ 0         │",
    "│ src/file-with-a-$-char.js │       flow │   100 % │     3 │       3 │ 0         │",
    "│ src/file-with-a-'-char.js │       flow │   100 % │     3 │       3 │ 0         │",
    "│ src/file-with-a-\\\\-char.js │       flow │   100 % │     3 │       3 │ 0         │",
    "│ src/file-with-a-\`-char.js │       flow │   100 % │     3 │       3 │ 0         │",
  ],
  "stderr": "",
}
`;
